File: project/ranyoung-onboarding/src/main/java/com/example/ranyoungonboarding/domain/Response.java
package com.example.ranyoungonboarding.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "responses")
@Getter
@Setter
public class Response {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id", nullable = false)
    private Survey survey;

    @OneToMany(mappedBy = "response", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Answer> answers = new ArrayList<>();

    @Column(nullable = false)
    private LocalDateTime submittedAt;

    @PrePersist
    protected void onCreate() {
        this.submittedAt = LocalDateTime.now();
    }

    // Helper method to maintain bidirectional relationship
    public void addAnswer(Answer answer) {
        answers.add(answer);
        answer.setResponse(this);
    }
}

File: project/minjong-onboarding/src/main/kotlin/com/innercircle/survey/domain/Response.kt
// src/main/kotlin/com/innercircle/survey/domain/Response.kt
package com.innercircle.survey.domain

import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "responses")
class Response(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    val survey: Survey,

    @Column(nullable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @OneToMany(mappedBy = "response", cascade = [CascadeType.ALL], orphanRemoval = true)
    val items: MutableList<ResponseItem> = mutableListOf()
) {
    // 응답 항목 추가 메서드
    fun addItem(item: ResponseItem) {
        items.add(item)
        item.response = this
    }
}